### ISSUE TITLE
Posting unicode-only tweets seems to break the bot

### ISSUE THREAD

**Issue by emmawinston** [mispy/twitter_ebooks#102]
Hello! I've been running a bot at https://twitter.com/botlovesyou for a while based on my own badly-handcoded Python, but have never been able to get it to consistently respond to mentions running on Heroku. twitter_ebooks seems to solve that, but the problem is when I try to run the program with a corpus containing just unicode characters (since botlovesyou's only vocabulary is emoji hearts) I get the following error:

```
@botlovesyou: #<NoMethodError: undefined method `each_with_index' for nil:NilClass>
	/Users/emmawinston/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/twitter_ebooks-3.1.5/lib/twitter_ebooks/suffix.rb:62:in `block in generate'
	/Users/emmawinston/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/twitter_ebooks-3.1.5/lib/twitter_ebooks/suffix.rb:59:in `upto'
	/Users/emmawinston/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/twitter_ebooks-3.1.5/lib/twitter_ebooks/suffix.rb:59:in `generate'
	/Users/emmawinston/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/twitter_ebooks-3.1.5/lib/twitter_ebooks/model.rb:256:in `make_statement'
	bots.rb:61:in `on_mention'
	/Users/emmawinston/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/twitter_ebooks-3.1.5/lib/twitter_ebooks/bot.rb:373:in `fire'
	/Users/emmawinston/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/twitter_ebooks-3.1.5/lib/twitter_ebooks/bot.rb:297:in `receive_event'
	/Users/emmawinston/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/twitter_ebooks-3.1.5/lib/twitter_ebooks/bot.rb:363:in `block in start'
	/Users/emmawinston/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/twitter-5.15.0/lib/twitter/streaming/client.rb:116:in `block in request'
	/Users/emmawinston/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/twitter-5.15.0/lib/twitter/streaming/response.rb:27:in `block in on_body'
	/Users/emmawinston/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/twitter-5.15.0/lib/twitter/streaming/response.rb:25:in `each'
	/Users/emmawinston/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/twitter-5.15.0/lib/twitter/streaming/response.rb:25:in `on_body'
	/Users/emmawinston/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/twitter-5.15.0/lib/twitter/streaming/response.rb:16:in `<<'
	/Users/emmawinston/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/twitter-5.15.0/lib/twitter/streaming/response.rb:16:in `<<'
	/Users/emmawinston/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/twitter-5.15.0/lib/twitter/streaming/connection.rb:22:in `stream'
	/Users/emmawinston/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/twitter-5.15.0/lib/twitter/streaming/client.rb:119:in `request'
	/Users/emmawinston/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/twitter-5.15.0/lib/twitter/streaming/client.rb:92:in `user'
	/Users/emmawinston/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/twitter_ebooks-3.1.5/lib/twitter_ebooks/bot.rb:362:in `start'
	/Users/emmawinston/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/twitter_ebooks-3.1.5/bin/ebooks:346:in `block (3 levels) in start'
	/Users/emmawinston/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/twitter_ebooks-3.1.5/bin/ebooks:344:in `loop'
	/Users/emmawinston/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/twitter_ebooks-3.1.5/bin/ebooks:344:in `block (2 levels) in start'
```

I note that when I consume the corpus it seems to think there are 0 statements, so I'm not sure if that has something to do with it. I am very, very much a Ruby beginner so apologies if there's a really obvious reason this is happening - any advice would be much appreciated. I have already required 'unicode' at the start of the bots.rb file.

**Comment by negatendo** [comment]
Yea the language modeling stuff only works with tokens larger than two characters, if I remember correctly. The error comes from having an empty model. 

I get where you are coming from though. I also use the twitter_ebooks framework for building other types of bots, and often ignore the text modeling stuff entirely. Is there any way you can read your emoji characters from a json file or something like that? Then you could simply tweet a sample from that data. There's probably no need for text modeling stuff with what you're trying to do.

**Comment by emmawinston** [comment]
I thought that was what I was doing, but maybe it's not? I've just put a list of emoji hearts into a file called 'heart.txt' and got it to consume that for the corpus, but it doesn't seem to recognise any of the text. Do I only need to consume the corpus if I'm doing language modelling? If not, how would I just get it to pull a single line from a text file? Sorry for my incredible lack of understanding, aargh.

**Comment by negatendo** [comment]
Yep! You don't need to fun the consume function on a text file for the data in the text file to appear inside of your bot. That feature is primarily meant for making combinations of words and phrasing out of text - aka the modeling. (And it expects longer text at the very least which is why it's breaking.)

You can just use Ruby to read your data. Here's an example to get you started: http://stackoverflow.com/questions/6012930/read-lines-of-a-file-in-ruby 

My suggested approach would be to create a separate file containing a Ruby class or function to read in your data and get a random sample. It will be easier to develop this outside of having to configure and run the bot. Then in bots.rb include your file and get your random emoji at tweet-time. Good luck!

**Comment by emmawinston** [comment]
Got it. Thank you. This was entirely down to error on my part. Closing, and I very, very much appreciate your help!

**Comment by mispy** [comment]
I think it's non-representative behavior not to include 1-length sentences anyway, even we if we can't recombine them. Fixed in 3.1.6.
